Projects/Market/Sources/PickMarket.swift

```diff
@@ -4,29 +4,86 @@ import hCore
 import hCoreUI
 
 public struct PickMarket: View {
-    let currentMarket: Market
+    var currentMarket: Market
     @PresentableStore var store: MarketStore
+    @State var code: String?
+
+    let onSave: ((Market) -> Void)?
+    let onCancel: (() -> Void)?
+
+    @State var selectedMarket: Market
 
     public init(
-        currentMarket: Market
+        currentMarket: Market,
+        onSave: ((Market) -> Void)? = nil,
+        onCancel: (() -> Void)? = nil
     ) {
         self.currentMarket = currentMarket
+        self.onSave = onSave
+        self.onCancel = onCancel
+        self.code = currentMarket.id
+        self.selectedMarket = currentMarket
     }
 
     public var body: some View {
         hForm {
-            hSection(Market.activatedMarkets, id: \.title) { market in
-                hRow {
-                    Image(uiImage: market.icon)
-                    Spacer().frame(width: 16)
-                    market.title.hText()
+            if onSave == nil {
```

Comment:
This section here is unnecessary. if onSave == nil should be here and body of 'if' also